app/controllers/ems_cloud_controller.rb

```diff
@@ -42,4 +42,61 @@ def restful?
 
   menu_section :clo
   has_custom_buttons
+
+  def sync_users
+    @ems = find_record_with_rbac(model, params[:id])
+    @in_a_form = true
+    drop_breadcrumb(:name => _("Sync Users"), :url => "/ems_cloud/sync_users")
+    @selected_admin_role = params[:admin_role]
+    @selected_member_role = params[:member_role]
+    @selected_password = params[:password]
+    @selected_verify = params[:verify]
+
+    if params[:cancel]
+      redirect_to(ems_cloud_path(params[:id]))
+      return
+    end
+
+    if params[:sync]
+      has_error = false
+      if @selected_password != @selected_verify
+        add_flash(_("Password/Confirm Password do not match"), :error)
+        has_error = true
+      end
+      if @selected_admin_role.blank?
+        add_flash(_("An admin role must be selected."), :error)
+        has_error = true
+      end
+      if @selected_member_role.blank?
+        add_flash(_("A member role must be selected."), :error)
+        has_error = true
+      end
+      if has_error
+        populate_sync_user_parameters
+      else
+        password_digest = nil
+        password_digest = BCrypt::Password.create(@selected_password) unless @selected_password.blank?
+        @ems.sync_users_queue(session[:userid], @selected_admin_role, @selected_member_role, password_digest)
+        redirect_to(ems_cloud_path(params[:id], :flash_msg => _("Sync users queued.")))
+      end
+    else
+      populate_sync_user_parameters
+    end
+  end
+
+  def populate_sync_user_parameters
+    @admin_roles = {}
+    @admin_roles["Choose Admin Role"] = nil
+    Rbac::Filterer.filtered(MiqUserRole).each do |r|
+      @admin_roles[r.name] = r.id
+    end
+
+    @member_roles = {}
+    @member_roles["Choose Member Role"] = nil
+    Rbac::Filterer.filtered(MiqUserRole).each do |r|
+      @member_roles[r.name] = r.id
+    end
```

Comment:
this one is the same as the one above, so why don't you `dup` it and add the special element after that?

Reply:
Ok.